### ISSUE TITLE
Don't raise S2368 which complains about multidimensional arrays when params keyword is declared on jagged arrays

### ISSUE THREAD

**Issue by markusschaber** [SonarSource/sonar-dotnet#4225]
### Description

The rule S2368 currently fires when the `params` keyword is used on jagged (nested) arrays.

However, at least for two-level jagged arrays declared with `params`, the caller of the function usually just experiences them as a variadic number of distinct arguments of the same type, so the warning should not be triggered in this case.

Additionally, the diagnostics should be fixed to stop calling jagged arrays "multidimensional", as multidimensional arrays (like `byte[,]`) are something different as jagged (nested) arrays (like `byte[][]`). 

### Repro steps

```
namespace SonarLintBugS2368
{
    using System;

    public static class GuidHelper
    {
        public static Guid CreateNameBasedGuid(params byte[][] nameComponents)
        {
            return Guid.Empty;
        }
    }
}
```

### Expected behavior

S2368 does not raise on this method.

### Actual behavior

S2368 is raised on this method, and additionally falsely claims the jagged array to be a multidimensional array.

### Known workarounds

Declare the method as private, and provide N public wrapper methods with 1, 2, 3, ... n parameters of type byte[].

### Related information

* C# Version 8.0 (but works with older versions, too).
* Visual Studio version 2019
* MSBuild / dotnet version: .NET 4.8 / .NET Core 3.1
* SonarScanner for .NET version (if used): SonarScanner for MSBuild 4.9/SonarScanner 4.3.0.2102 (Also reproducible with SonarLint for VS 2019 Version 4.33.1.28913)
* Operating System: Windows 10

**Comment by csaba-sagi-sonarsource** [comment]
Hi @markusschaber thanks for the feedback. I can confirm the false positive.
